### PR TITLE
Add Rate limiter

### PR DESCRIPTION
**Pull request by trajan0x**
## Description

Fixes # (issue)

## Checklist

- [ ] New Contracts have been tested
- [ ] Lint has been run
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: contracts/bridge/RateLimiter.sol [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+    function addToRetryQueue(bytes32 kappa, bytes memory toRetry)
+        external
+        onlyRole(BRIDGE_ROLE)
+    {
+        rateLimited.set(kappa, toRetry);
+    }
+
+    function retryByKappa(bytes32 kappa) external onlyRole(LIMITER_ROLE) {
+        (bytes memory toRetry, ) = rateLimited.get(kappa);
```
Comment (ChiTimesChi) [inline contracts/bridge/RateLimiter.sol:184]: Check if kappa already exists before calling bridge. I think assumption that **all kappas** in the queue were not yet processed is too strong to preserve.
```solidity
if (!IBridge(BRIDGE_ADDRESS).kappaExists(kappa)) {
    (bool success, bytes memory returnData) = BRIDGE_ADDRESS.call(toRetry);
    require(
        success,
        Strings.append("could not call bridge:", _getRevertMsg(returnData))
    );
}
rateLimited.remove(kappa);
```

File: contracts/bridge/RateLimiter.sol [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    }
+
+    function retryCount(uint8 count) external onlyRole(LIMITER_ROLE) {
+        // no issues casting to uint8 here. If length is greater then 255, min is always taken
+        uint8 attempts = uint8(
+            MathUpgradeable.min(uint256(count), rateLimited.length())
+        );
+
+        for (uint8 i = 0; i < attempts; i++) {
+            (bytes32 kappa, bytes memory toRetry, ) = rateLimited.at(i);
```
Comment (ChiTimesChi) [inline contracts/bridge/RateLimiter.sol:200]: See comment above:
https://github.com/synapsecns/synapse-contracts/pull/114/files#r849141913

Also, some generalization on retrying the kappas is required.

File: contracts/bridge/RateLimiter.sol [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+                success,
+                Strings.append(
+                    "could not call bridge for kappa: ",
+                    Strings.toHex(kappa),
+                    " reverted with: ",
+                    _getRevertMsg(returnData)
+                )
+            );
+
+            rateLimited.remove(kappa);
```
Comment (ChiTimesChi) [inline contracts/bridge/RateLimiter.sol:214]: Doing `map.remove()` in the same for loop as `map.at()` will break things. Either do two separate loops for iteration and deletion, or change `Map` to queue and use `peek` and `poll` to get the first element and then delete it.

**Comment by trajan0x** [comment]
Moving to #116 to follow our git flow

**Pull request closed without merging**